osu.Game/Database/EFToRealmMigrator.cs

```diff
@@ -236,8 +270,8 @@ private void migrateScores(DatabaseWriteUsage db)
                                 HasReplay = ((IScoreInfo)score).HasReplay,
                                 Date = score.Date,
                                 PP = score.PP,
-                                BeatmapInfo = realm.All<BeatmapInfo>().First(b => b.Hash == score.BeatmapInfo.Hash),
-                                Ruleset = realm.Find<RulesetInfo>(score.Ruleset.ShortName),
+                                BeatmapInfo = beatmap,
+                                Ruleset = ruleset,
```

Comment:
Not sure if it's a big problem but these are specified twice now (once here, once in ctor). Same goes for `User` really but I can't easily highlight it on this diff.

In contrast, `migrateBeatmaps` looks to have moved stuff from properties to ctor entirely where possible.